doc/user_guide/usage/run.rst

```diff
@@ -9,22 +9,25 @@ Pylint is meant to be called from the command line. The usage is ::
 
    pylint [options] modules_or_packages
 
-By default the ``pylint`` command only accepts a list of python modules and packages. Using a
-directory which is not a package results in an error::
+By default the ``pylint`` command only accepts a list of python modules and packages.
+On versions below 2.15, specifying a directory that is not an explicit package
+(with ``__init__.py``) results in an error::
 
     pylint mydir
     ************* Module mydir
     mydir/__init__.py:1:0: F0010: error while code parsing: Unable to load file mydir/__init__.py:
     [Errno 2] No such file or directory: 'mydir/__init__.py' (parse-error)
 
-When ``--recursive=y`` option is used, modules and packages are also accepted as parameters::
+When the ``--recursive=y`` option is used, modules and packages are also accepted as parameters::
 
     pylint --recursive=y mydir mymodule mypackage
 
 This option makes ``pylint`` attempt to discover all modules (files ending with ``.py`` extension)
-and all packages (all directories containing a ``__init__.py`` file).
+and all explicit packages (all directories containing a ``__init__.py`` file).
+As of version 2.15, ``pylint`` will also discover namespace packages but there are some edge
+cases still under development (where you might still find ``--recursive=y`` useful).
```

Comment:
I don't quite understand this sentence. From the wording it sounds like it is (in most cases) no longer necessary to use `--recursive=y`, yet the preceding sentences all describe the usage *with* it.

Reply:
Improved comprehensibility in [63d654c](https://github.com/PyCQA/pylint/pull/7153/commits/63d654cf55c911a44d5ecb8ef122c52040f1b483)